Application/Services/PostService.cs

```diff
@@ -0,0 +1,131 @@
+                .Include(p => p.User)
+                .Where(p => !p.IsPrivate)
+                .Select(p => _mapper.Map<Post, FeedPostVm>(p))
+                .ToListAsync();
+
+            return posts;
+        }
+
+        public async Task<List<PrivatePostVm>> GetPrivatePostsByUser()
+        {
+            var user = await _userManager.GetUserAsync(_httpContextAccessor.HttpContext.User);
+            var posts = await _postRepository.GetAll()
+                .Where(p => p.IsPrivate && p.UserId == user.Id)
+                .Select(p => _mapper.Map<Post, PrivatePostVm>(p))
+                .ToListAsync();
+
+            return posts;
+        }
+
+        public async Task<List<BlogPostVm>> GetPublicPostsByUser(Guid userId)
+        {
+            var posts = await _postRepository.GetAll()
+                .Where(p => p.UserId == userId && !p.IsPrivate)
+                .Select(p => _mapper.Map<BlogPostVm>(p))
+                .ToListAsync();
+
+            return posts;
+        }
+
+        public async Task<PostDetailsVm> GetPost(Guid id)
+        {
+            var user = await _userManager.GetUserAsync(_httpContextAccessor.HttpContext.User);
+
+            var post = await _postRepository.GetById(id)                
+                .Include(q => q.Comments.OrderBy(c => c.CreatedOn)).ThenInclude(q => q.User)
+                .Include(p => p.User)
+                .Select(p => _mapper.Map<PostDetailsDto>(p))
+                .SingleOrDefaultAsync();
+
+            if (post == null)
+                throw new NotFoundException(nameof(Post), id);
+
+            var postVm = _mapper.Map<PostDetailsVm>(post);            
+
+            if(post.UserId == user.Id)
+                postVm.IsCanEdit = true;
+
+            return postVm;
+        }
+
+        public async Task Create(ExtendedCreatePostDto createPostDto)
+        {
+            var post = _mapper.Map<Post>(createPostDto);
+            post.Preview = TruncateText(post.Message, NumberValues.PostPreviewLength);
+
+            await _postRepository.AddAsync(post);
+            await _postRepository.SaveChangesAsync();
+        }
+
+        public async Task<ValidationResult> CreateComment(CreateCommentDto createCommentDto)
+        {
+            if (string.IsNullOrEmpty(createCommentDto.Message))
+            {
+                return new ValidationResult(
+                ErrorMessages.CommentEmpty,
+                new[] { nameof(PostDetailsDto.Comments) });
+            }
+
+            var comment = _mapper.Map<Comment>(createCommentDto);
+            await _postRepository.CreateComment(comment);
+            await _postRepository.SaveChangesAsync();
+
+            return null;
+        }
+
+        public async Task<Post> UpdatePost(UpdatePostDto dto)
+        {
+            var post = await _postRepository.GetById(dto.Id).SingleOrDefaultAsync();
+            post.Message = dto.Message;
+            post.Preview = TruncateText(post.Message, NumberValues.PostPreviewLength);
```

Comment:
I think this is not a great way to do text preview. You can't find a `PostPreviewLength` value that will look good on all screen sizes. Better to do this on the client (browser)